molr-mole-core/src/main/java/io/molr/mole/core/runnable/lang/ForeachBranchRootMapped.java

```diff
@@ -39,21 +41,21 @@ private void createAndAddForeachBlock() {
 		builder().forEachBlock(block, itemsPlaceholder, itemPlaceholder, transformedItemPlaceholder, function);
 	}
 	
-	public OngoingForeachBranch<U> branch(String name, Placeholder<?>... placeholders) {
+	public OngoingContextualBranch<U> branch(String name, Placeholder<?>... placeholders) {
 		createAndAddForeachBlock();
 		for (int i = 0; i < placeholders.length; i++) {
 			if(placeholders[i].equals(Placeholders.LATEST_FOREACH_ITEM_PLACEHOLDER)) {
 				placeholders[i]=itemPlaceholder;
```

Comment:
I am not so sure if it is clean in all cases to mutate the incoming array....? Potentially do a defensive copy first?